src/Core/Domain/Contact/DTO/EditableContact.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Contact\DTO;
+
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\ContactException;
+use PrestaShop\PrestaShop\Core\Domain\Contact\ValueObject\ContactId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Email;
+
+/**
+ * Class EditableContact is responsible for providing required data form contact form.
+ */
+class EditableContact
+{
+    /**
+     * @var ContactId
+     */
+    private $contactId;
+
+    /**
+     * @var array|string[]
+     */
+    private $localisedTitles;
+
+    /**
+     * @var Email|null
+     */
+    private $email;
+
+    /**
+     * @var bool
+     */
+    private $isMessagesSavingEnabled;
+
+    /**
+     * @var string[]
+     */
+    private $localisedDescription;
+
+    /**
+     * @var int[]
+     */
+    private $shopAssociation;
+
+    /**
+     * @param int $contactId
+     * @param string[] $localisedTitles
+     * @param string $email
+     * @param bool $isMessagesSavingEnabled
+     * @param string[] $localisedDescription
+     * @param int[] $shopAssociation
+     *
+     * @throws ContactException
+     * @throws DomainConstraintException
+     */
+    public function __construct(
+        $contactId,
+        array $localisedTitles,
+        $email,
+        $isMessagesSavingEnabled,
+        $localisedDescription,
+        array $shopAssociation
+    ) {
+        $this->contactId = new ContactId($contactId);
```

Comment:
@matks I remember you told me that I should use value objects inside  - is it true all in all cases?

Reply:
This was in order to follow CQRS rule that is quite nicely stated by https://stackoverflow.com/a/15900638/8071080:
"A command is data-oriented, not object-oriented. Its properties should be mostly primitives. You're not modeling anything here, just sending flat data around.

Using custom types forces you to publish the types to every client that wants to use the commands. It will also prevent you from changing these types in the future without some hassle.

Then there's seriazability: The less custom types the simpler.

In messages like commands and events I'd go with primitives as far as reasonably possible. Duplication is not the issue here, you want low coupling, serializability, change tolerance."

Reply:
For DTOs we dont have this limit.